### DISCUSSION TITLE
Change how initial data is imported?

### DISCUSSION THREAD

**Discussion by jbergstroem** [jbergstroem/mariadb-alpine#140] [Polls]
In [this issue](https://github.com/jbergstroem/mariadb-alpine/issues/19) I have explored the option to remove installing a mysql client for loading initial data. Not having a client available will not work for most decently sized imports:
- passing data via `stdin` has [a limit of 60,000 bytes](https://github.com/MariaDB/server/blob/945a5439cc17f6375117522602fdf3b29d9f85c2/sql/sql_bootstrap.h#L26)
- passing data via `--init-file` has [a limit of 20,000 bytes](https://github.com/MariaDB/server/blob/945a5439cc17f6375117522602fdf3b29d9f85c2/sql/sql_bootstrap.h#L31)

The reason I'm not a fan is that we connect to the internet to get this done. I see three options forward (see below) What do you think is the best option?

My picks would be either no change or add a mysql client; it can be useful for other reasons (tests, improved health check, fully no-root runtime, convenience).

**Comment by jbergstroem** [comment]
I've been exploring building a simpler mysql import cli to stream data to mysql, but I have a hard time gaining significant space savings. The mariadb client is about 3Mb and my basic rust cli is ~2.4Mb which I don't find significant enough. I'd like it to be closer to 0.5Mb; basically speak the wire protocol and pass sql as-is.

**Comment by williamdes** [comment]
I think having a client in a server image is what you expect to interact with the server when needed. Privileges, data, ...

**Reply by jbergstroem** [reply]
Thanks for your feedback!

**Reply by jbergstroem** [reply]
Question! As much as I agree on the convenience end; what do you mean with requirements in the context of privileges or data? The client would only have access through the database interface and not necessarily via disk.

You would more or less start two instances of `jbergstroem/mariadb-alpine` instead of one (coupled with `jbergstroem/mariadb-client-alpine`).

**Reply by williamdes** [reply]
Two images could be nice to have for sure. It's just useful to have a cli when you have some problematic event that requires you to login and do some root account hacking for example

**Reply by jbergstroem** [reply]
> Two images could be nice to have for sure

I actually already have `jbergstroem/mariadb-client-alpine` available. It is being used in the test suite.

> It's just useful to have a cli when you have some problematic event that requires you to login and do some root account hacking for example

What benefit is it to have the mysql cli in the same container? What can you do that you cannot via another container?

**Comment by jbergstroem** [comment]
Just created a PR to reflect the outcome of this discussion: https://github.com/jbergstroem/mariadb-alpine/pull/177

**Reply by jbergstroem** [reply]
Since this is now merged, I will close the discussion.
